costmap_2d/plugins/static_layer.cpp

```diff
@@ -150,6 +158,9 @@ unsigned char StaticLayer::interpretValue(unsigned char value)
 
 void StaticLayer::incomingMap(const nav_msgs::OccupancyGridConstPtr& new_map)
 {
+  if (first_map_only_)
+    first_map_.reset(new nav_msgs::OccupancyGrid(*new_map));
```

Comment:
this would seem to make every map the first map? Only by stopping and starting costmap updates would the map_sub_ be shutdown and the first_map actually be locked in